contracts/mainnet/connectors/morpho-blue/helpers.sol

```diff
@@ -0,0 +1,190 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+import {Id, IMorpho, MarketParams, Position, Market} from "./interfaces/IMorpho.sol";
+import "../../common/stores.sol";
+import "../../common/basic.sol";
+import "../../common/interfaces.sol";
+import {MorphoBalancesLib} from "./libraries/periphery/MorphoBalancesLib.sol";
+import {UtilsLib} from "./libraries/UtilsLib.sol";
+import {MarketParamsLib} from "./libraries/MarketParamsLib.sol";
+import {SharesMathLib} from "./libraries/SharesMathLib.sol";
+
+abstract contract Helpers is Stores, Basic {
+    using MorphoBalancesLib for IMorpho;
+    using MarketParamsLib for MarketParams;
+    using UtilsLib for uint256;
+    using SharesMathLib for uint256;
+
+    IMorpho public constant MORPHO_BLUE =
+        IMorpho(0x777777c9898D384F785Ee44Acfe945efDFf5f3E0); // TODO: Update
+
+    uint256 internal constant MARKET_PARAMS_BYTES_LENGTH = 5 * 32;
+
+    /// @dev The number of virtual assets of 1 enforces a conversion rate between shares and assets when a market is
+    /// empty.
+    uint256 internal constant VIRTUAL_ASSETS = 1;
+
+    /// @dev The number of virtual shares has been chosen low enough to prevent overflows, and high enough to ensure
+    /// high precision computations.
+    uint256 internal constant VIRTUAL_SHARES = 1e6;
+
+    enum Mode {
+        Collateral,
+        Repay,
+        Supply
+    }
+
+    /// @notice Handles Eth to Weth conversion if assets are provided.
+    function _performEthToWethConversion(
+        MarketParams memory _marketParams,
+        uint256 _assets,
+        address _onBehalf,
+        uint256 _getId,
+        Mode _mode
+    ) internal returns (Id _id, MarketParams memory, uint256 _amt) {
+        _amt = getUint(_getId, _assets);
+
+        bool _isModeCollateral = _mode == Mode.Collateral;
+
+        bool _isEth = _isModeCollateral
+            ? _marketParams.collateralToken == ethAddr
+            : _marketParams.loanToken == ethAddr;
+
+        _marketParams = updateTokenAddresses(_marketParams);
+
+        _id = _marketParams.id();
+
+        // Check for max value
+        if (_assets == type(uint256).max) {
+            uint256 _maxBalance = _isEth
+                ? address(this).balance
+                : _isModeCollateral
+                ? TokenInterface(_marketParams.collateralToken).balanceOf(
+                    address(this)
+                )
+                : TokenInterface(_marketParams.loanToken).balanceOf(
+                    address(this)
+                );
+
+            if (_mode == Mode.Repay) {
+                uint256 _amtDebt = getPaybackBalance(
+                    _id,
+                    _marketParams,
+                    _onBehalf
+                );
+
+                _amt = UtilsLib.min(_maxBalance, _amtDebt);
+            } else {
+                _amt = _maxBalance;
+            }
+        }
+
+        // Perform eth to weth conversion if necessary
+        convertEthToWeth(
+            true,
```

Comment:
Following: https://github.com/Instadapp/dsa-connectors/pull/321#discussion_r1428849200

I believe I went too fast on this: formerly, the value of `_isEth` was actually only checked once, because the bool value passed as argument of `convertEthToWeth` is `true`
So IMO either revert the change and check for `_isEth`, or pass `_isEth` as arg of `convertEthToWeth`

I personally think the latter is better

Reply:
Passing `_isEth` as the argument!